src/main/scala/interfaces/Verification.scala

```diff
@@ -7,13 +7,52 @@
 package viper.silicon.interfaces
 
 import viper.silicon.interfaces.state.Chunk
+import viper.silicon.state.terms.Term
 import viper.silicon.state.{State, Store}
 import viper.silver.verifier.{Counterexample, Model, VerificationError}
-import viper.silicon.state.terms.Term
 
 /*
  * Results
  */
+class VerificationResultWrapper(verificationResultsarg: Seq[VerificationResult]) {
```

Comment:
A few things bug me about this class, and I have the impression that a better design is possible when taking immutability into account. `&&`, `getFatals`, `getFailures` and `containsFatal` all filter/partition the results, which seems potentially redundant if many calls are made.

Since parameter `verificationResultsarg` is an immutable sequence of immutable results, it would suffice to partition the incoming results into failures and successes, and to store the two separately. This should reduce the number of subsequent filter steps necessary in aforementioned methods.

Moreover, all methods but `&&` could be turned into `lazy val`s instead of `def`s, since subsequent calls should always return the same results anyway.